test/e2e/config_test.go

```diff
@@ -689,6 +690,152 @@ rules:
 	})
 }
 
+func TestConfigHavingRegex(t *testing.T) {
+	yaml1 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+  annotations:
+    foo1: bar1
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+  annotations:
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+`
+
+	yaml2 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+data:
+  player_initial_lives: "3"
+`
+
+	configYaml := `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+  - path: [metadata, annotations, {regex: "^foo"}]
+    type: %s
+    sources: [new, existing]
+    resourceMatchers:
+      - apiVersionKindMatcher: {apiVersion: v1, kind: ConfigMap}
+`
+
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	fieldsExcludedInMatch := []string{"kapp.k14s.io/app", "creationTimestamp:", "resourceVersion:", "uid:", "selfLink:", "kapp.k14s.io/association"}
+	name := "test-config-path-regex"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("creating an app with multiple resources", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--diff-changes-yaml"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		expectedOutput := `
+---
+# create: configmap/game-demo (v1) namespace: kapp-test
+apiVersion: v1
+data:
+  player_initial_lives: "3"
+kind: ConfigMap
+metadata:
+  annotations:
+    foo1: bar1
+    foo2: bar2
+  labels:
+  name: game-demo
+  namespace: kapp-test
+---
+# create: configmap/game-test (v1) namespace: kapp-test
+apiVersion: v1
+data:
+  player_initial_lives: "3"
+kind: ConfigMap
+metadata:
+  annotations:
+    foo2: bar2
+  labels:
+  name: game-test
+  namespace: kapp-test
+`
+		out = strings.TrimSpace(replaceTarget(replaceSpaces(replaceTs(out))))
+		out = clearKeys(fieldsExcludedInMatch, out)
+
+		expectedOutput = strings.TrimSpace(replaceSpaces(expectedOutput))
+		require.Contains(t, out, expectedOutput, "output does not match")
```

Comment:
Not sure what the test case is trying to achieve.

Reply:
This is just to verify that both configmaps are getting created successfully.

Reply:
If you all think it's not required I can remove it and just have the below two cases

Reply:
Yeah, seems like an unnecessary step in this test.

Reply:
Adopted.